.github/workflows/check-kernel-config.yml

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# This action checks that required kernel configs have not been removed or
+# modified to an undesirable value. It also checks that new configs added to
+# the kernel config files are in the list of required configs.
+name: Kernel Required Configs Check
+
+on:
+  push:
+    branches: [main, dev, 1.0*, 2.0*]
+  pull_request:
+    branches: [main, dev, 1.0*, 2.0*]
```

Comment:
We might benefit from [filtering based on changed files](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore). That would allow us to get rid of all of the changed file logic, and prevent this from running on unrelated PRs.

Reply:
I had no idea that's a thing. Sounds like a great idea, I need to update other PR checks now.;)

Reply:
I should say- you can only get rid of all the changed file logic if you have a static set of files you're looking for. For, say, the linter, where we need to know what specific subset of specs to scan, we still need it.

Reply:
I played around with this for other PR checks. It looks like required checks will block a PR instead of being skipped when we use `paths` but none of the config files were changes for a given PR. I haven't had time to see how to configure this better, so that irrelevant PR checks are skipped correctly, but unless we do that, we cannot make this PR check a required one (it would have a similar effect as the linter check).